tests/src/tests/links/linksTests.js

```diff
@@ -0,0 +1,441 @@
+      });
+    });
+
+    it('fail to create long dynamic link with unsupported android parameters', () => {
+      return new Promise((resolve, reject) => {
+        const success = tryCatch(() => {
+          // Assertion
+          reject(new Error('createDynamicLink did not fail.'));
+        }, reject);
+
+        const failure = tryCatch((error) => {
+          // Assertion
+          error.code.includes('links/failure').should.be.true();
+          resolve();
+        }, reject);
+
+        const data = {
+          dynamicLinkInfo: {
+            link,
+            dynamicLinkDomain,
+            androidInfo: {
+              androidPackageName,
+              someInvalidParameter: 'invalid',
+              someOtherParameter: 'invalid',
+            },
+            iosInfo: {
+              iosBundleId,
+            },
+          },
+        };
+
+        // Test
+
+        links.createDynamicLink(data)
+        .then(success)
+        .catch(failure);
+      });
+    });
+
+    it('fail to create long dynamic link with unsupported social parameters', () => {
+      return new Promise((resolve, reject) => {
+        const success = tryCatch(() => {
+          // Assertion
+          reject(new Error('createDynamicLink did not fail.'));
+        }, reject);
+
+        const failure = tryCatch((error) => {
+          // Assertion
+          error.code.includes('links/failure').should.be.true();
+          resolve();
+        }, reject);
+
+        const data = {
+          dynamicLinkInfo: {
+            link,
+            dynamicLinkDomain,
+            androidInfo: {
+              androidPackageName,
+            },
+            iosInfo: {
+              iosBundleId,
+            },
+            socialMetaTagInfo: {
+              someInvalidParameter: 'invalid',
+              someOtherParameter: 'invalid',
+            },
+          },
+        };
+
+        // Test
+
+        links.createDynamicLink(data)
+        .then(success)
+        .catch(failure);
+      });
+    });
+
+    it('create short (unguessable) dynamic link with all supported parameters', async () => {
+      const url = 'https://www.google.co.il/search?q=react+native+firebase';
+      const data = {
+        dynamicLinkInfo: {
+          link: url,
+          dynamicLinkDomain,
+          androidInfo: {
+            androidPackageName,
+            androidFallbackLink,
+            androidMinPackageVersionCode,
+          },
+          iosInfo: {
+            iosBundleId,
+            iosFallbackLink,
+            iosCustomScheme,
+            iosIpadFallbackLink,
+            iosIpadBundleId,
+            iosAppStoreId,
+          },
+          socialMetaTagInfo: {
+            socialTitle,
+            socialDescription,
+            socialImageLink,
+          },
+        },
+      };
+
+      const result = await links.createShortDynamicLink(data);
+      result.should.startWith(`https://${dynamicLinkDomain}`);
+
+      const response = await fetch(result);
```

Comment:
This introduces a linter error. You should add `fetch` as a global under `.eslintrc`